pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go

```diff
@@ -237,7 +239,59 @@ func TestTransitiveClosure(t *testing.T) {
 			},
 			want: wantGraph{ids: []string{"a", "b", "c"}},
 		},
-		// TODO(bowei): add error test cases.
+		{
+			name:  "nodes with error results in err",
+			cloud: mockCloud,
+			graph: func() *rgraph.Builder {
+				g := rgraph.NewBuilder()
+				addNodeToGraph(g, "a", []string{"b", "c"})
+				addNode("b", []string{"d", "e"}, nodeIsInError)
+				addNode("d", []string{"a"})
```

Comment:
Do we need these additional nodes if we are testing the error cases? We should make the error cases simple so people who maintain this later don't have to figure out whether or not the extra setup was part of the test? This comment applies to all of the test cases being added.